apps/src/aiComponentLibrary/chatMessage/ChatMessage.tsx

```diff
@@ -61,29 +65,54 @@ const ChatMessage: React.FunctionComponent<ChatMessageProps> = ({
     }
   }, [chatMessageText, role, status, showProfaneUserMessage]);
 
+  // All this to check the width of the chat message and set the width of the
+  // div holding the children appropriately.
+  const chatMessageRef = useRef<HTMLDivElement>(null);
+  const [chatMessageWidth, setChatMessageWidth] = useState<number>(0);
+  const resizeObserver = useMemo(() => {
+    return new ResizeObserver(() => {
+      if (chatMessageRef.current !== null) {
+        setChatMessageWidth(chatMessageRef.current.offsetWidth);
+      }
+    });
+  }, [chatMessageRef]);
+
+  useEffect(() => {
+    if (chatMessageRef.current !== null) {
+      resizeObserver.observe(chatMessageRef.current);
+    }
+  }, [chatMessageRef, resizeObserver]);
```

Comment:
Slick! I know @molly-moen and @thomasoniii have been thinking about resizing panels etc for Upper Lab. Is there a generalizable solution for resizing things that's worth sharing?